src/AbstractSQLRules2SQL.ts

```diff
@@ -889,6 +892,17 @@ const typeRules: Dictionary<MatchFn> = {
 		const replacement = TextValue(getAbstractSqlQuery(args, 2), indent);
 		return `REPLACE(${str}, ${find}, ${replacement})`;
 	},
+	ExtractJSONPathAsText: (args, indent) => {
+		checkMinArgs('ExtractJSONPathAsText', args, 1);
+		if (engine !== Engines.postgres) {
+			throw new SyntaxError(
+				'ExtractJSONPathAsText not supported on: ' + engine,
+			);
+		}
+		const column = TextValue(getAbstractSqlQuery(args, 0), indent);
+		const path = args[1] ? getAbstractSqlQuery(args, 1)[1] : '';
```

Comment:
`getAbstractSqlQuery(args, 1)[1]` feels hacky...

Reply:
Yeah, really it should be something like
```suggestion
		const column = TextArrayValue(getAbstractSqlQuery(args, 1), indent);
```
since `{foo}` is just an array literal and can be supplied in different forms, eg
```sql
SELECT '{"foo":{"bar":"baz"}}'::json #>> ARRAY_AGG(t.path), ARRAY_AGG(t.path)
FROM (
	SELECT *
	FROM (
		SELECT 1 x, 'foo' as path
		UNION
		SELECT 2 x, 'bar' as path
	) t
	ORDER BY t.x
) t;
-- "baz"	"{foo,bar}"

SELECT '{"foo":{"bar":"baz"}}'::json #>> t.path, t.path
FROM (
	SELECT ARRAY['foo','bar'] as path
) t;
-- "baz"	"{foo,bar}"
```
postgres docs: https://www.postgresql.org/docs/14/functions-json.html#FUNCTIONS-JSON-OP-TABLE
```
json #>> text[] → text
jsonb #>> text[] → text
```